### ISSUE TITLE
The induction tactic does too much reduction

### ISSUE THREAD

**Issue by QGarchery** [Deducteam/lambdapi#861]
Hi everone !

With version 2.1.0, lambdapi used to verify the content of [tmr.txt](https://github.com/Deducteam/lambdapi/files/8504782/tmr.txt). It's no longer the case with version 2.2.0. The problem is that the `induction` tactic performs reductions on the subgoals it generates.

Motivation for fixing this issue is similar to #693. Note also that, in issue #693, `whnf` was mentioned  and that this function was changed in 2.2.0.

PS: when I manage to port my use case to 2.2.0, I would love to add it as an example in the `tests/OK` folder.

**Comment by fblanqui** [comment]
@gabrielhdt This seems to be related somehow to the changes you made in #819 . Could you please check this?

**Comment by gabrielhdt** [comment]
It seems the culprit reduction happens in `Unif.solve_noexn` 
https://github.com/Deducteam/lambdapi/blob/d264cd8ed859635dc0f71e1d9213a911b46939b6/src/handle/tactic.ml#L87

Instrumenting the code with
```diff
@@ -84,8 +84,10 @@ let tac_solve : popt -> proof_state -> proof_state = fun pos ps ->
   in
   p := {!p with metas = List.fold_left f MetaSet.empty gs_typ
               ; to_solve = List.rev_map get_constr gs_unif};
+  Format.eprintf "Problem before %a" problem p;
   if not (Unif.solve_noexn p) then
     fatal pos "Unification goals are unsatisfiable.";
+  Format.eprintf "Problem after %a" problem p;
   (* remove in [gs_typ] the goals that have been instantiated, and simplify
      the others. *)
   let not_instantiated = function
```
I get the output (trimmed for clarity)
```
Problem before { recompute=false; metas={(?7:(?6:TYPE)), (?6:TYPE),
(?5:(?4:TYPE)), (?4:TYPE), (?3:(?2:TYPE)), (?2:TYPE)};
to_solve=[Π x: ℕ, π ((?3:(?2:TYPE)) x) ≡ Π x: ℕ, π (x = x);
(?6:TYPE) ≡ Π x0: ℕ, π ((?3:(?2:TYPE)) x0) → π ((?3:(?2:TYPE)) (s x0));
(?4:TYPE) ≡ π ((?3:(?2:TYPE)) zero); (?2:TYPE) ≡ ℕ → Prop];
 unsolved=[] }
Problem after { recompute=true;
metas={(?7:Π x0: ℕ, π (`∀ₜₚ P, P x0 ⇒ P x0) → π (`∀ₜₚ P, P (s x0) ⇒ P (s x0))),
(?5:π (`∀ₜₚ P, P zero ⇒ P zero))}; to_solve=[];
 unsolved=[] }
[tact] (?5:π (`∀ₜₚ P, P zero ⇒ P zero)): π (`∀ₜₚ P, P zero ⇒ P zero)
       reflexivity
```

**Comment by fblanqui** [comment]
Fixed in #863.
